kafka-rest/src/main/java/io/confluent/kafkarest/controllers/SchemaManagerImpl.java

```diff
@@ -154,15 +154,22 @@ private RegisteredSchema getSchemaFromSchemaVersion(
 
     ParsedSchema parsedSchema;
     try {
+      Optional<ParsedSchema> opSchema = schemaProvider.parseSchema(schema, /* isNew= */ false);
```

Comment:
@rayokota Weirdly enough - 

line 158 & 163 - shows OpSchema is empty, contains null.
BUT line 162 & 166, show OpSchema is NOT empty. This is super weird!!! (**Why is this happening?**)

Since `orElseThrows`, at line 166, DOESN'T consider the optional as non-null, it doesn't throw at line 166, which earlier the test used to. And then it returns NullPointerException at line 171.

[text.txt](https://github.com/confluentinc/kafka-rest/files/10579766/text.txt)
Attaching log output from running the test.

Reply:
Ok, instead of changing the test, let's change`SchemaManagerImpl`.  On line 146, after getting the `schema`, have `SchemaManagerImpl` throw an exception if `schema.getSchema()` is null.

Reply:
@rayokota its done.